src/lib.rs

```diff
@@ -381,4 +381,49 @@ impl Contract {
 
         lockup_ids
     }
+
+    pub fn discard_draft_group(&mut self, draft_group_id: DraftGroupIndex) {
+        self.assert_deposit_whitelist(&env::predecessor_account_id());
+
+        let mut draft_group = self
+            .draft_groups
+            .get(&draft_group_id as _)
+            .expect("draft group not found");
+        draft_group.discard();
+
+        if draft_group.draft_indices.is_empty() {
+            self.draft_groups.remove(&draft_group_id as _);
+        } else {
+            self.draft_groups.insert(&draft_group_id as _, &draft_group);
+        }
+    }
+
+    pub fn delete_drafts(&mut self, draft_ids: Vec<DraftIndex>) {
+        let mut draft_group_lookup: HashMap<DraftGroupIndex, DraftGroup> = HashMap::new();
```

Comment:
Missing permissions check.

Reply:
After the draft group is discarded, all the drafts in the group are destined to be removed, and it doesn't really matter who does it.
It's similar to convert_drafts, where we allow converting them to after the group is funded.

Reply:
@alexaunter I agree with your reasoning, though to improve the situation I suggest you add an explicit comment at the beginning of the function to clearly state why there is no need in the permissions check and point to the assert below that checks whether it is allowed to remove the draft